linkerd/app/core/src/accept_error.rs

```diff
@@ -32,12 +33,14 @@ where
     }
 
     fn call(&mut self, sock: T) -> Self::Future {
-        self.0.accept(sock).then(|v| {
-            if let Err(e) = v {
-                let error: Error = e.into();
-                tracing::debug!(%error, "Connection failed");
-            }
-            Ok(())
-        })
+        Box::new(self.0.accept(sock).map(|connection_future| {
+            Box::new(connection_future.then(|v| {
+                if let Err(e) = v {
+                    let error: Error = e.into();
+                    tracing::debug!(%error, "Connection failed");
+                }
+                Ok(())
+            })) as Self::Response
+        })) as Self::Future
```

Comment:
Are these `as` directives load-bearing?

Regardless, I think in this case, we should prefer concrete future types to Boxed types. Given this is such a fundamental part of the data path, we should avoid unnecessary alloc when it's easy.

Reply:
Yeah, it looks like there is only ever one possible `Future` type returned here, and one possible `Response` type for that future?